cogs/birthday/birthday.py

```diff
@@ -512,52 +543,100 @@ async def setSelfBirthday(self, ctx: Context, *, birthday: MonthDayConverter):
+                        f"{headerBad}: You took too long. Not setting your birthday."
+                    )
+                    return
 
-            def check(msg: discord.Message):
-                return msg.author == ctx.author and msg.channel == ctx.author.dm_channel
+                if response.content.lower() != "yes":
+                    await destination.send(f"{headerBad}: Declined. Not setting your birthday.")
+                    return
+                # Set birthday and notify user that their birthday has been set
+                await birthdayConfig.get_attr(KEY_BDAY_MONTH).set(birthday.month)
+                await birthdayConfig.get_attr(KEY_BDAY_DAY).set(birthday.day)
+                await birthdayConfig.get_attr(KEY_ADDED_BEFORE).set(True)
+
+                await destination.send(
+                    f"{headerGood}: Successfully set your birthday to "
+                    f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}.",
+                    delete_after=SENSITIVE_MSG_TTL,
+                )
 
-            try:
-                response = await self.bot.wait_for("message", timeout=30.0, check=check)
-            except asyncio.TimeoutError:
-                # hide the birthday portion within the previous message
-                await ctx.author.send(
-                    f"{headerBad}: You took too long. Not setting your birthday."
+                self.logger.info(
+                    "%s#%s (%s) added their birthday as %s",
+                    ctx.author.name,
+                    ctx.author.discriminator,
+                    ctx.author.id,
+                    birthdayStr,
                 )
-                return
 
-            if response.content.lower() != "yes":
-                await ctx.author.send(f"{headerBad}: Declined. Not setting your birthday.")
+                # explicitly check to see if the role should be applied to the user
+                # if the month and day just so happen to be the same as it is now.
+                await self.checkBirthday()
                 return
 
-            await birthdayConfig.get_attr(KEY_BDAY_MONTH).set(birthday.month)
-            await birthdayConfig.get_attr(KEY_BDAY_DAY).set(birthday.day)
-            await birthdayConfig.get_attr(KEY_ADDED_BEFORE).set(True)
+            try:
+                await mainFlow(ctx.author, carefree=True)
+                return
+            except discord.Forbidden:
+                await ctx.send(
+                    "\n".join(
+                        (
+                            "You have disabled DMs from this server. Would you "
+                            "still like to continue here? All messages containing your "
+                            "birthday will be deleted after a short delay.",
+                            f"Type {bold('`yes`', escape_formatting=False)} to confirm.",
+                        )
+                    )
+                )
```

Comment:
micronit: due to how `black` works, we get a bunch of lines that have *only* single parentheses, which are pretty ugly honestly LOL. That's not anything big deal, but we can extract this string out to a variable like how we do `confirmationStr`, just so when reading this portion of code, we don't get distracted by 8 lines that have only 4 important lines...

So somewhere outside, maybe before this `try`-`except`, we have
```py
            confirmationStr = "\n".join(
                (
                    "You have disabled DMs from this server. Would you "
                    "still like to continue here? All messages containing your "
                    "birthday will be deleted after a short delay.",
                    f"Type {bold('`yes`', escape_formatting=False)} to confirm.",
                )
            )
```

And then we have here at this line:

```suggestion
                await ctx.send(confirmationStr)
```

Up to you to name it but we can also just have `confirmationStr` because we don't need the content of the *previous* `confirmationStr` anymore.

Reply:
I have extracted the string and named it `noDmStr`